### PR TITLE
Obsolete rag_bloody, migrate to rag (filthy)

### PR DESCRIPTION
**Pull request by FuelType-Memes**
#### Summary

SUMMARY: Content "Obsolete rag_bloody, migrate to rag (filthy)"

#### Purpose of change

I don't think we need a special "bloody rag" item when we have `FILTHY` flag that represents organic matter. This issue was also touched upon in #42568 and #42574

#### Describe the solution

Obsolete `rag_bloody` migrate to `rag` with an added `FILTHY` flag

#### Describe alternatives you've considered

Leaving as is

#### Testing

Game runs fine

#### Additional context

Would be great if we could define sprites based on item flags

### CODE REVIEW

File: data/json/items/obsolete.json [hunk 1]
```diff
@@ -2755,5 +2755,29 @@
...
+    "symbol": ",",
+    "color": "red",
+    "name": { "str": "blood soaked rag" },
+    "description": "A large rag, drenched in blood.  It could be cleaned with boiling water.",
+    "material": [ "cotton" ],
+    "flags": [ "NO_SALVAGE", "TRADER_AVOID" ],
+    "weight": "80 g",
+    "volume": "250 ml"
+  },
+  {
```
Comment (Fris0uman) [inline data/json/items/obsolete.json:2771]: obsolete recipe should move to there https://github.com/CleverRaven/Cataclysm-DDA/blob/master/data/json/recipes/recipe_obsolete.json

File: data/json/recipes/recipe_obsolete.json [hunk 2]
```diff
@@ -2499,5 +2499,17 @@
...
+  {
+    "type": "recipe",
+    "result": "rag",
+    "category": "CC_OTHER",
+    "subcategory": "CSC_OTHER_MATERIALS",
+    "skill_used": "survival",
+    "time": "3 m",
+    "autolearn": true,
+    "tools": [ [ [ "surface_heat", 3, "LIST" ] ] ],
+    "components": [ [ [ "water", 1 ], [ "water_clean", 1 ] ], [ [ "rag_bloody", 1 ] ] ],
```
Comment (anothersimulacrum) [inline data/json/recipes/recipe_obsolete.json:2512]: None of this should be here anymore.
```suggestion
```

File: data/json/items/obsolete.json [hunk 3]
```diff
@@ -2750,6 +2750,19 @@
...
+    "type": "GENERIC",
```
Comment (ZhilkinSerg) [inline data/json/items/obsolete.json:2765]:
```suggestion
```

File: data/json/mapgen/fema/FEMA_le_03.json [hunk 4]
```diff
@@ -42,7 +42,7 @@
         "L": { "item": "surgery", "chance": 30 },
         "X": { "item": "hospital_lab", "chance": 80 }
       },
-      "item": { "B": { "item": "rag_bloody", "amount": [ 0, 5 ] } },
+      "item": { "B": { "item": "rag", "amount": [ 0, 5 ], "custom-flags": [ "FILTHY" ] } },
```
Comment (ZhilkinSerg) [inline data/json/mapgen/fema/FEMA_le_03.json:45]: "custom-flags" is not working here.

Comment (FuelType-Memes) [reply data/json/mapgen/fema/FEMA_le_03.json:45]: what is the proper way to do it then?

Comment (ZhilkinSerg) [reply data/json/mapgen/fema/FEMA_le_03.json:45]: Add code that it would be supported in mapgen objects.

Comment (FuelType-Memes) [reply data/json/mapgen/fema/FEMA_le_03.json:45]: This whole PR is completely JSON, seems to me that actual cpp edits should be handled outside of this

File: data/json/recipes/recipe_others.json [hunk 5]
```diff
@@ -1230,7 +1230,7 @@
   },
   {
     "type": "recipe",
-    "activity_level": "MODERATE_EXERCISE",
+    "activity_level": "LIGHT_EXERCISE",
```
Comment (ZhilkinSerg) [inline data/json/recipes/recipe_others.json:1233]:
```suggestion
    "activity_level": "MODERATE_EXERCISE",
```

File: data/json/mapgen/fema/FEMA_le_03.json (hunk 4 above)
Comment (ZhilkinSerg) [reply data/json/mapgen/fema/FEMA_le_03.json:45]: I've added code to support this in #44362. After it is merged, you can rebase your PR on top of latest master and CI error should go away.

**Pull request merged**